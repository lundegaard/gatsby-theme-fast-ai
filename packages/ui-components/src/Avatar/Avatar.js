import React from 'react';
import PropTypes from 'prop-types';

import mapResponsiveProperty from '../utils/mapResponsiveProperty';
import * as Types from '../types';
import Box from '../Box';

const addPx = mapResponsiveProperty(x => `${x}px`);

const Avatar = ({ size = 35, sx, ...rest }) => (
	<Box
		display="block"
		sx={{
			backgroundColor: 'white',
			width: size,
			height: size,
			lineHeight: addPx(size),
			textAlign: 'center',
			fontWeight: 'bold',
			borderRadius: 9999,
			borderWidth: '1px',
			borderStyle: 'solid',
			borderColor: 'transparent',
			...sx,
		}}
		{...rest}
	/>
);

Avatar.propTypes = {
	size: Types.responsiveNumber,
	sx: PropTypes.object,
};

export default Avatar;
