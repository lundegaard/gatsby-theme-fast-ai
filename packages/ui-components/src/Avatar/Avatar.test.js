import React from 'react';
import { shallow } from 'enzyme';

import Avatar from './Avatar';

describe('<Avatar />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Avatar {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
