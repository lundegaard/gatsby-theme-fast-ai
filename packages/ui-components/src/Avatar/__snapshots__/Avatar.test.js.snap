// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> matches snapshot when shallowly rendered 1`] = `
<Box
  display="block"
  sx={
    Object {
      "backgroundColor": "white",
      "borderColor": "transparent",
      "borderRadius": 9999,
      "borderStyle": "solid",
      "borderWidth": "1px",
      "fontWeight": "bold",
      "height": 35,
      "lineHeight": Array [
        "35px",
      ],
      "textAlign": "center",
      "width": 35,
    }
  }
/>
`;
